### ISSUE TITLE
Add `.url` getter to `RouteInfo`

### ISSUE THREAD

**Issue by buschtoens** [emberjs/rfcs#658]
Currently generating a URL for a `RouteInfo` object requires quite a bit of manual, repetitive plumbing:

```ts
getURLFromRouteInfo(router: RouterService, routeInfo: RouteInfo) {
  const { name, paramNames, params, queryParams } = routeInfo;
  const orderedParams: string[] = paramNames.map(n => params[n]);
  const url = router.urlFor(name, ...orderedParams, { queryParams });
  return url;
}
```

I would like to expose this as a `get url()` directly on `RouteInfo` instead.

I haven't looked deeply into the code for `RouteInfo` yet, but I guess that it's basically a passthrough of [`router.js/lib/router/route-info.ts`](https://github.com/tildeio/router.js/blob/master/lib/router/route-info.ts) somewhere, so it may possibly lack an owner, which could make things a bit more complicated, but maybe it would also allow to completely avoid using the `RouterService` in the first place.

Would this feature be appreciated? Does it require a fully-fledged RFC?

**Comment by buschtoens** [comment]
I've published [`ember-routing-utils`](https://github.com/buschtoens/ember-routing-utils), which includes `getURLFromRouteInfo(routeInfo)` and `removeRootURL(url)` to fix a `transitionTo()` quirk https://github.com/emberjs/ember.js/issues/18422#issuecomment-538668443.

**Comment by NullVoxPopuli** [comment]
I seem to need utilities like this once every few months. maybe we just pr to the router service and discuss implementation there?

**Comment by wagenet** [comment]
Are any of you interested in writing an RFC for this?

**Comment by NullVoxPopuli** [comment]
do we really need a full RFC (+ core review) for "just adding a property"?

**Comment by wagenet** [comment]
@NullVoxPopuli could just make the PR and see if it gets merged 😄

**Comment by NullVoxPopuli** [comment]
Excellent -- I added this to my list of things: https://github.com/users/NullVoxPopuli/projects/5/views/1

**Comment by wagenet** [comment]
Corollary is that if for some reason it becomes clear that it's not an easy merge and that more discussion is needed, it should be obvious what needs to be covered by an associated RFC.

**Comment by NullVoxPopuli** [comment]
agreed -- gonna try to tackle this as part of work -- it'll allow us to delete a few hundred lines of code in total, I think (due to duplication)
